sphinx_immaterial/apidoc_formatting.py

```diff
@@ -355,16 +360,17 @@ def _builder_inited(app: sphinx.application.Sphinx) -> None:
 
     registry = get_object_description_option_registry(app)
     options_map: Dict[str, List[Tuple[int, Dict[str, Any]]]] = {}
-    options_patterns: List[Tuple[Pattern, int, Dict[str, Any]]] = []
+    options_patterns: List[Tuple[Pattern[str], int, Dict[str, Any]]] = []
 
     default_options = {}
     for name, registered_option in registry.items():
         default_options[name] = registered_option.default
 
     # Validate options
+    print("validating options")
     for i, (pattern, options) in enumerate(
         pydantic.parse_obj_as(
-            List[Tuple[Pattern, Dict[str, Any]]],
+            List[Tuple[Pattern, Dict[str, Any]]],  # type: ignore
```

Comment:
Could change `Dict[str, Any]` to `ObjectDescriptionOptions`.

I guess `type: ignore` is because pydantic doesn't support `Pattern[str]` ?

Reply:
> I guess `type: ignore` is because pydantic doesn't support `Pattern[str]` ?

Correct. It reports something about not being able to handle `re.compile()`. I also tried `AnyStr`, but it still failed.